appengine/go11x/pubsub/authenicated_push/main.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+	"sync"
+
+	"google.golang.org/api/idtoken"
+	"google.golang.org/api/option"
+)
+
+var (
+	// Messages received by this instance.
+	messagesMu sync.Mutex
+	messages   [][]byte
```

Comment:
Other tutorial apps have started using a `type app struct { ... }` to hold variables that would otherwise be globals. HTTP handlers then need to be defined on that type. Could we do that here?

Reply:
Done.